### PR TITLE
Shape ID are not uniq, only the combination (shape_id, pt_sequence)

### PR DESCRIPTION
**Pull request by antoine-de**
This fixes https://github.com/etalab/transport-validator/pull/181 that wrongly said that the shapes were duplicated whereas they had different pt_sequences.

I choose to only display the first 10 duplicated pt_sequence of a shape to reduce the noise for large duplicated shapes, but this can be discussed.

sorry for the noise @AntoineAugusti

### CODE REVIEW

File: src/validators/raw_gtfs.rs [hunk 1]
```diff
@@ -18,18 +20,57 @@ fn check_duplicates<O: gtfs_structures::Id + gtfs_structures::Type>(
...
+                .or_insert_with(|| vec![])
+                .push(shape.sequence);
+        }
+        ids.insert(id);
+    }
+
+    shape_issues
+        .into_iter()
+        .map(|(shape_id, duplicate_sequences)| {
+            let max_displayed_dup = 10;
```
Comment (Tristramg) [inline src/validators/raw_gtfs.rs]: Maybe put it as a constant a bit higher in the code, as it more of a configuration?

File: src/validators/raw_gtfs.rs [hunk 2]
```diff
@@ -18,18 +20,57 @@ fn check_duplicates<O: gtfs_structures::Id + gtfs_structures::Type>(
...
+            let more_dup = if duplicate_sequences.len() > max_displayed_dup {
+                "..."
+            } else {
+                ""
+            };
+            let d = format!(
+                "Shape has duplicated pt_sequence: {}{}",
+                duplicate_sequences.iter().take(10).join(","),
```
Comment (Tristramg) [inline src/validators/raw_gtfs.rs]: use the constant here
`join(", ")` is prettier ;)

File: src/validators/raw_gtfs.rs [hunk 3]
```diff
@@ -18,18 +20,57 @@ fn check_duplicates<O: gtfs_structures::Id + gtfs_structures::Type>(
...
+        }
+        ids.insert(id);
+    }
+
+    shape_issues
+        .into_iter()
+        .map(|(shape_id, duplicate_sequences)| {
+            let max_displayed_dup = 10;
+            let more_dup = if duplicate_sequences.len() > max_displayed_dup {
+                "..."
```
Comment (AntoineAugusti) [inline src/validators/raw_gtfs.rs]:
```suggestion
                "…"
```

File: test_data/duplicates/shapes.txt [hunk 4]
```diff
@@ -0,0 +1,30 @@
+shape_id,shape_pt_lat,shape_pt_lon,shape_pt_sequence,shape_dist_traveled
```
Comment (AntoineAugusti) [inline test_data/duplicates/shapes.txt:1]: Do we have a test file (or maybe we can include it here) where we have no duplicates (with `(shape_id, shape_pt_sequence)`) but multiple lines for the same `shape_id`

File: src/validators/raw_gtfs.rs [hunk 5]
```diff
@@ -18,18 +20,57 @@ fn check_duplicates<O: gtfs_structures::Id + gtfs_structures::Type>(
...
+            } else {
+                ""
+            };
+            let d = format!(
+                "Shape has duplicated pt_sequence: {}{}",
```
Comment (Tristramg) [inline src/validators/raw_gtfs.rs:50]:
```suggestion
                "Shape {shape_id} has duplicated pt_sequence: {}{}",
```

File: src/validators/raw_gtfs.rs (hunk 1 above)
Comment (AntoineAugusti) [inline src/validators/raw_gtfs.rs]: Move this constant outside the loop?

File: src/validators/raw_gtfs.rs (hunk 5 above)
Comment (antoine-de) [reply src/validators/raw_gtfs.rs:50]: The name is already in the issue no? But as you wish

File: test_data/duplicates/shapes.txt (hunk 4 above)
Comment (antoine-de) [reply test_data/duplicates/shapes.txt:1]: There is in the [test file](https://github.com/etalab/transport-validator/blob/4c23a1814ad565df90a4b2dcc698721bbf245951/test_data/duplicates/shapes.txt), B_shp has 2 points for different pt_sequences, and A_shp has 4 points, with only 2 being duplicated

File: src/validators/raw_gtfs.rs (hunk 5 above)
Comment (Tristramg) [reply src/validators/raw_gtfs.rs:50]: oh sorry, I didn’t that far. Indeed no need to duplicate!

File: test_data/duplicates/shapes.txt (hunk 4 above)
Comment (AntoineAugusti) [reply test_data/duplicates/shapes.txt:1]: Thanks! 🙏 I overlooked that 🙈

File: src/validators/raw_gtfs.rs (hunk 3 above)
Comment (antoine-de) [reply src/validators/raw_gtfs.rs]: :+1:

File: test_data/duplicates/shapes.txt (hunk 4 above)
Comment (antoine-de) [reply test_data/duplicates/shapes.txt:1]: no worry, it's nice to check thoroughly this, I did not realize there were potential alerting email on mistake on the validator :sweat_smile:

**Comment by antoine-de** [comment]
Thanks for all the reviews I tried to take everything into account

**Review by Tristramg — APPROVED** [review]

**Review by AntoineAugusti — APPROVED** [review]
Thanks for the quick fix @antoine-de!

**Pull request merged**